### DISCUSSION TITLE
Why Debian 11 (not 12)

### DISCUSSION THREAD

**Discussion by spusuf** [Jip-Hop/jailmaker#28] [Q&A]
The linux container image server linked hosts both Debian 11 (bullseye) and Debian 12 (Bookworm). Is there a reason for using Debian 11? Compatibility issues you encountered? Has Debian 12 even been tested?

**Comment by Jip-Hop** [comment]
If you'd like to test 12 and report back that would be appreciated. I currently recommend to use the same debian version inside the jail as outside. So with the new release in October I'll probably start recommending 12.

**Reply by Kieros** [reply]
I tried it but I keep running into this issue

sudo apt-get update
Ign:1 http://deb.debian.org/debian bookworm InRelease
Ign:2 http://deb.debian.org/debian-security bookworm-security InRelease
Ign:3 http://deb.debian.org/debian bookworm-updates InRelease
Ign:1 http://deb.debian.org/debian bookworm InRelease
Ign:2 http://deb.debian.org/debian-security bookworm-security InRelease
Ign:3 http://deb.debian.org/debian bookworm-updates InRelease
Ign:1 http://deb.debian.org/debian bookworm InRelease
Ign:2 http://deb.debian.org/debian-security bookworm-security InRelease
Ign:3 http://deb.debian.org/debian bookworm-updates InRelease
Err:1 http://deb.debian.org/debian bookworm InRelease
  Temporary failure resolving 'deb.debian.org'
Err:2 http://deb.debian.org/debian-security bookworm-security InRelease
  Temporary failure resolving 'deb.debian.org'
Err:3 http://deb.debian.org/debian bookworm-updates InRelease
  Temporary failure resolving 'deb.debian.org'
Reading package lists... Done
W: Failed to fetch http://deb.debian.org/debian/dists/bookworm/InRelease  Temporary failure resolving 'deb.debian.org'
W: Failed to fetch http://deb.debian.org/debian-security/dists/bookworm-security/InRelease  Temporary failure resolving 'deb.debian.org'
W: Failed to fetch http://deb.debian.org/debian/dists/bookworm-updates/InRelease  Temporary failure resolving 'deb.debian.org'
W: Some index files failed to download. They have been ignored, or old ones used instead.


I have been looking at the source.list
but I even used the official. from debian wiki. It keeps doing this.

deb http://deb.debian.org/debian bookworm main
deb-src http://deb.debian.org/debian bookworm main

deb http://deb.debian.org/debian-security/ bookworm-security main
deb-src http://deb.debian.org/debian-security/ bookworm-security main

deb http://deb.debian.org/debian bookworm-updates main
deb-src http://deb.debian.org/debian bookworm-updates main

When I go to deb.debian.org it resolves just fine.
Any clue?

**Reply by spusuf** [reply]
can you ping deb.debian.org from the jail's shell?

**Comment by Jip-Hop** [comment]
Looks like a DNS problem. What does /etc/resolv.conf look like?Op 21 aug. 2023 om 03:19 heeft spusuf ***@***.***> het volgende geschreven:﻿
can you ping deb.debian.org from the jail's shell?

**Reply by Kieros** [reply]
I can ping it 
![image](https://github.com/Jip-Hop/jailmaker/assets/99768682/32cdc1ac-2aa7-4c7b-98c8-49187b3f37ea)

Also the resolv.conf file contains my dns server the router pfsense in this case

**Comment by Kieros** [comment]
I have the same issue with the recommended debian install.
I can ping the deb.debian.org from within the jail.
But when updating it says it can't resolve
This started after renaming the zpool. Could it be of any ifluence?
It is weird that I can ping it right?

**Reply by Kieros** [reply]
I found it.
By accident I changed the dataset jailmaker in a ACL
So I removed the unix permissions I guess.
Is there a way to access it via SMB?

I got the docker engine installed and running

**Comment by spusuf** [comment]
Debian 12 works flawlessly. Confirmed myself.

**Comment by spusuf** [comment]
Just want to bump this and say I'm on RC1 and it's is running on Debian 12. Is your plan to switch over to Debian 12 as default when the full truenas release comes out, now, arbitrarily later, or implement a script that checks the underlying OS using something like 'lsb_release -a' and matches that?

As discussed Debian 12 jails work on Debian 11 based TrueNAS so forwards compatibility isnt an issue, but there are many packages that arent available for Debian 11 (such as php8 and a few others that I cant recall) so Debian 12 would be appreciated.

**Reply by Jip-Hop** [reply]
You can already deviate from the default using the jailmaker 'install wizard' and choose debian 12. I'll make debian 12 the default when I've upgraded to the 23.10 release and I've manually confirmed jailmaker works.

**Reply by Kieros** [reply]
Yes I have been using it since I used this working all fine.

**Comment by qudiqudi** [comment]
So running Debian 12 now inside jails is not a problem whatsoever, as documented here and by others.
How about updating existent Debian 11 jails to 12? Afaik, there is no underlying function in jlmkr itself to assist. Question remains:
Is it a bumpy road, should we stick to "create new and migrate over" or can it be achieved within the existing jails like a simple "apt dist-upgrade"?

**Reply by Jip-Hop** [reply]
I did “create new and migrate over” but my docker jail was very simple. I don’t think there’s anything in jailmaker or systemd-nspawn which would prevent you from following the Debian upgrade steps: https://wiki.debian.org/DebianUpgrade

**Reply by qudiqudi** [reply]
Just did that on a smaller docker jail and it seems to play out nicely.
